kotsadm/pkg/kurl/kurl_nodes.go

```diff
@@ -0,0 +1,148 @@
+package kurl
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/kotsadm/pkg/k8s"
+	"github.com/replicatedhq/kots/kotsadm/pkg/logger"
+	v1 "k8s.io/api/core/v1"
+	kuberneteserrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type KurlNodes struct {
+	Nodes         []Node `json:"nodes"`
+	HA            bool   `json:"ha"`
+	IsKurlEnabled bool   `json:"isKurlEnabled"`
+}
+
+type Node struct {
+	Name           string            `json:"name"`
+	IsConnected    bool              `json:"isConnected"`
+	CanDelete      bool              `json:"canDelete"`
+	KubeletVersion string            `json:"kubeletVersion"`
+	CPU            CapacityAvailable `json:"cpu"`
+	Memory         CapacityAvailable `json:"memory"`
+	Pods           CapacityAvailable `json:"pods"`
+	Conditions     NodeConditions    `json:"conditions"`
+}
+
+type CapacityAvailable struct {
+	Capacity  float64 `json:"capacity"`
+	Available float64 `json:"available"`
+}
+
+type NodeConditions struct {
+	MemoryPressure bool `json:"memoryPressure"`
+	DiskPressure   bool `json:"diskPressure"`
+	PidPressure    bool `json:"pidPressure"`
+	Ready          bool `json:"ready"`
+}
+
+// GetNodes will get a list of nodes with stats
+func GetNodes(client kubernetes.Interface) (*KurlNodes, error) {
+	nodes, err := client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "list nodes")
+	}
+
+	mc, err := k8s.Metricsset()
+	if err != nil {
+		return nil, errors.Wrap(err, "get metrics client")
+	}
+
+	toReturn := KurlNodes{}
+
+	for _, node := range nodes.Items {
+		cpuCapacity := CapacityAvailable{}
+		memoryCapacity := CapacityAvailable{}
+		podCapacity := CapacityAvailable{}
+
+		memoryCapacity.Capacity = float64(node.Status.Capacity.Memory().Value()) / 1000000000 // capacity in GB
+
+		cpuCapacity.Capacity, err = strconv.ParseFloat(node.Status.Capacity.Cpu().String(), 64)
```

Comment:
I assume these are all type quantity. Why convert to string?

Reply:
If there's a better way to convert to float, I'd love to hear it
Or I'll just try returning the decimal type

Reply:
I thought maybe you could get it from AsInt64() but idk

Reply:
Hmmm
.String() gives a value of `8` in my testing thus far
Maybe AsInt64() would give 8000 instead?